### PR TITLE
Fixes #3855

### LINKED ISSUE #3855: Update Intro YouTube Tensors & Autograd tutorials — deprecated in-place ops, matmul alias, formatting
## Description

The [Introduction to PyTorch Tensors](https://docs.pytorch.org/tutorials/beginner/introyt/tensors_deeper_tutorial.html) and [The Fundamentals of Autograd](https://docs.pytorch.org/tutorials/beginner/introyt/autogradyt_tutorial.html) tutorials contain deprecated APIs and outdated patterns that should be modernized.

### Deprecated APIs (both files)

- **`torch.sin_(b)` → `b.sin_()`**: Module-level in-place functions (`torch.sin_`, `torch.cos_`, etc.) were deprecated in PyTorch 2.1. Use the tensor method `.sin_()` instead.

### Outdated patterns (both files)

- **`torch.linalg.matmul()` → `torch.matmul()`**: `torch.linalg.matmul` is a documented alias, but `torch.matmul` is the canonical name used in all PyTorch docs. Using the alias may confuse readers looking up the API.
- **`.format()` → f-strings**: Replace old-style string formatting with f-strings (PEP 498).
- **Update in-place ops documentation text**: The tutorial text says in-place arithmetic is on `torch.Tensor`, "not attached to the `torch` module like many other functions (e.g., `torch.sin()`)". This contrast no longer applies after fixing `torch.sin_(b)` → `b.sin_()`. Update the surrounding explanation.

## Files

- [`beginner_source/introyt/tensors_deeper_tutorial.py`](https://github.com/pytorch/tutorials/blob/main/beginner_source/introyt/tensors_deeper_tutorial.py)
- [`beginner_source/introyt/autogradyt_tutorial.py`](https://github.com/pytorch/tutorials/blob/main/beginner_source/introyt/autogradyt_tutorial.py)

cc @subramen

### PR DESCRIPTION
**Pull request by jvz37**
Fixes https://github.com/pytorch/tutorials/issues/3855 
## Description
- Replace `torch.sin_(b)` with `b.sin_()` in `tensors_deeper_tutorial.py`
- Replace `torch.sin_(a)` with `a.sin_()` in `autogradyt_tutorial.py`
- Replace `torch.linalg.matmul()` with `torch.matmul()` in `tensors_deeper_tutorial.py`
- Update editorial text around in-place ops in `tensors_deeper_tutorial.py`

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @subramen

### CODE REVIEW

**Comment by sekyondaMeta** [comment]
@claude review and compare to work in issue #3855

File: beginner_source/introyt/autogradyt_tutorial.py [hunk 1]
```diff
@@ -452,57 +460,57 @@ def add_tensors2(x, y):
...
     run_on_gpu = True
-    
+
 x = torch.randn(2, 3, requires_grad=True)
 y = torch.rand(2, 3, requires_grad=True)
 z = torch.ones(2, 3, requires_grad=True)
 
 with torch.autograd.profiler.profile(use_cuda=run_on_gpu) as prf:
     for _ in range(1000):
         z = (z / x) * y
```
Comment (svekars) [inline beginner_source/introyt/autogradyt_tutorial.py:495]:
```suggestion
device = torch.accelerator.current_accelerator() if torch.accelerator.is_available() else torch.device("cpu")

x = torch.randn(2, 3, requires_grad=True)
y = torch.rand(2, 3, requires_grad=True)
z = torch.ones(2, 3, requires_grad=True)

with torch.autograd.profiler.profile(device.type == "cuda") as prf:
    for _ in range(1000):
        z = (z / x) * y
```

Comment (jvz37) [reply beginner_source/introyt/autogradyt_tutorial.py:495]: Hi! I tried to commit your suggestion but got: 'Applying suggestions on deleted lines is currently not supported'. Should I apply the change manually or wait for your input?

**Comment by sekyondaMeta** [comment]
@jvz37 any updates on this?

**Review by svekars — APPROVED** [review]
